lib/saluki-components/src/sources/checks/builder/python/python_modules.rs

```diff
@@ -1,29 +1,71 @@
 use std::sync::OnceLock;
 
 use pyo3::prelude::*;
+use pyo3::types::PyDict;
+use saluki_config::GenericConfiguration;
+use saluki_context::tags::TagSet;
+use saluki_core::data_model::event::{service_check::ServiceCheck, Event};
 use saluki_error::{generic_error, GenericError};
 use tokio::sync::mpsc::Sender;
 use tracing::{debug, error, trace};
 
 use crate::sources::checks::check_metric::{CheckMetric, MetricType};
 
 // Global state to store the sender
-static METRIC_SENDER: OnceLock<Sender<CheckMetric>> = OnceLock::new();
+static METRIC_SENDER: OnceLock<Sender<Event>> = OnceLock::new();
+// Global state to store the configuration
+static GLOBAL_CONFIGURATION: OnceLock<GenericConfiguration> = OnceLock::new();
+// Global state to store the configuration
+static GLOBAL_HOST: OnceLock<String> = OnceLock::new();
 
 /// Sets the metric sender to be used by the aggregator module.
-pub fn set_metric_sender(check_metrics_tx: Sender<CheckMetric>) -> &'static Sender<CheckMetric> {
+pub fn set_metric_sender(check_metrics_tx: Sender<Event>) -> &'static Sender<Event> {
     METRIC_SENDER.get_or_init(|| check_metrics_tx)
 }
 
+/// Stores the configuratioon for integration to use in the Python module.
+pub fn set_configuration(configuration: Option<GenericConfiguration>) -> &'static GenericConfiguration {
+    GLOBAL_CONFIGURATION.get_or_init(move || configuration.unwrap())
+}
+
+/// Stores the configuratioon for integration to use in the Python module.
+pub fn set_hostname(hostname: String) -> &'static String {
+    GLOBAL_HOST.get_or_init(|| hostname)
+}
+
 fn try_send_metric(metric: CheckMetric) -> Result<(), GenericError> {
     match METRIC_SENDER.get() {
         Some(sender) => sender
-            .try_send(metric)
+            .try_send(metric.into())
             .map_err(|e| generic_error!("Failed to send metric: {}", e)),
         None => Err(generic_error!("Metric sender not initialized.")),
     }
 }
 
+fn try_send_service_check(service_check: ServiceCheck) -> Result<(), GenericError> {
+    match METRIC_SENDER.get() {
+        Some(sender) => sender
+            .try_send(Event::ServiceCheck(service_check))
+            .map_err(|e| generic_error!("Failed to send metric: {}", e)),
+        None => Err(generic_error!("Metric sender not initialized.")),
+    }
+}
+
+fn get_config_key(key: String) -> String {
+    match GLOBAL_CONFIGURATION.get() {
+        Some(configuration) => configuration.get_typed_or_default::<String>(&key),
+
+        None => "".to_string(),
+    }
+}
+
+fn fetch_hostname() -> String {
+    match GLOBAL_HOST.get() {
+        Some(host) => host.clone(),
+        None => "".to_string(),
+    }
+}
```

Comment:
This _could_ return `&'static str` since it's effectively static now.

```suggestion
fn fetch_hostname() -> &'static str {
    match GLOBAL_HOST.get() {
        Some(host) => host.as_str(),
        None => "",
    }
}
```